### DISCUSSION TITLE
uplift runtime error

### DISCUSSION THREAD

**Discussion by markalston-edf** [gembaadvantage/uplift#444] [General]
Hello :) 

I am trying to add uplift to a project, but keep getting the following error when I run `uplift bump`:

```
• checking git             
   • next semantic version    
      • largest increment detected from commits increment=Minor
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x100fa853c]

goroutine 1 [running]:
github.com/gembaadvantage/uplift/internal/task/nextsemver.Task.Run({}, 0x1400025ac08)
        /home/runner/work/uplift/uplift/internal/task/nextsemver/nextsemver.go:75 +0x33c
github.com/gembaadvantage/uplift/internal/task.Execute(0x1400025ac08, {0x1400002bb98?, 0x1400023e040?, 0x1011572a8?})
        /home/runner/work/uplift/uplift/internal/task/runner.go:72 +0x178
main.bumpFiles({{0x0?, 0xea09d2de00000000?}, 0x1400023e040?}, {0x1011572a8?, 0x1400011c020?})
        /home/runner/work/uplift/uplift/cmd/uplift/bump.go:102 +0x118
main.newBumpCmd.func1(0x140001aea00?, {0x10143ac20?, 0x4?, 0x101009129?})
        /home/runner/work/uplift/uplift/cmd/uplift/bump.go:72 +0x34
github.com/spf13/cobra.(*Command).execute(0x14000127808, {0x10143ac20, 0x0, 0x0})
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:985 +0x840
github.com/spf13/cobra.(*Command).ExecuteC(0x14000126608)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1117 +0x344
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1041
main.main()
        /home/runner/work/uplift/uplift/cmd/uplift/main.go:11 +0x34
```

My .uplift.yml file looks like this:
```
bumps:
  - file: ./package.json
    regex:
      - pattern: '"version": "$VERSION"'
        semver: true
        count: 1
```

and my package.json looks like this:
```
{
  "version": "5.1.0",
  ....
}
```
Any ideas why I'm getting this?

Thank you :)

**Comment by ga-andrew-b** [comment]
Hi :)

The panic looks like a nil pointer dereference, I'm not sure why we aren't checking for possible errors (that would cause the pointer to be nil) in a few places:
- https://github.com/gembaadvantage/uplift/blob/main/internal/task/nextsemver/nextsemver.go#L71
- https://github.com/gembaadvantage/uplift/blob/main/internal/task/nextsemver/nextsemver.go#L45

My very quick read of the code suggests it might be due to the way we find the latest tag (latestTag()).  Have you got some tags on this repo that would match the pattern "*.*.*", e.g. "very.important.commit", or "Test0.1.2" i.e. a non semantic versioned tag but does include periods?  

If you are able to build the tool, you could a add some error returns instead of ignoring them on the lines I've linked to above as that may give you some more info.

**Reply by markalston-edf** [reply]
Thank you Andrew :) Yes the previous tag is `V5.0.13`. I'm guessing we shouldn't be including the V prefix in the tag name? Will sort that out and let you know if it fixes the issue!

**Reply by markalston-edf** [reply]
Removing the V prefix from the previous tag resolve the issue - thank you very much :)

**Reply by ga-andrew-b** [reply]
No problem.  Tags with a preceding 'v' are allowed (even if it is not a true semantic version), in fact I think they are required for Go repos otherwise the versioning within the Go ecosystem does not work (you cannot go get a version) e.g. `go get github.com/some-repo@v5.0.13`.  But, as in your case, a capital 'V' is not allowed.
